connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/executable/InboundExecutableRegistryImpl.java

```diff
@@ -236,20 +237,24 @@ private boolean matchesQuery(RegisteredExecutable executable, ActiveExecutableQu
               .stream()
               .map(InboundConnectorElement::element)
               .toList();
+          case InvalidDefinition invalid -> invalid.data().connectorElements().stream()
+              .map(InboundConnectorElement::element)
+              .toList();
         };
     var type =
         switch (executable) {
           case Activated activated -> activated.context().getDefinition().type();
-          case FailedToActivate failed -> failed.data().type();
+          case FailedToActivate failed -> failed.data().connectorElements().getFirst();
           case ConnectorNotRegistered notRegistered -> notRegistered.data().type();
+          case InvalidDefinition invalid -> invalid.data().connectorElements().getFirst().type();
         };
 
     return elements.stream()
         .anyMatch(
             element ->
                 query.bpmnProcessId() == null
```

Comment:
Consider extracting this 7-line conditional into its own method.

I could imagine that for a developer used to this codebase, this conditional is very clear. But for new readers, it's a little overwhelming aha 😄

Reply:
Lol good point, I split it into a few methods for readability